tests/load/bigquery/test_bigquery_table_builder.py

```diff
@@ -863,6 +869,16 @@ def sources() -> List[DltResource]:
         assert not no_hints_table.description
         assert hints_table.description == "Once upon a time a small table got hinted."
 
+    # run sources() again to make sure that table hints are ignored on alter table
```

Comment:
Reading this in isolation suggests that the intended behaviour is to not perform the ALTER migration for OPTION hint.

I suggest we leave the original `test_adapter_additional_table_hints_table_description` test and add `test_adapter_additional_table_hints_table_description_with_alter_table` as separate one, with an assert like

```python
assert hints_table.description == "Once upon a time a small table got hinted twice.",
``` 

We can then mark this test as `skip`  until support is added.
Let me know what you think!